### PR TITLE
[SHAD-338]Log HOST_APP_OPEN and RECCO_SDK_OPEN events

### PR DESCRIPTION
**Pull request by robertofrontado**
### Jira
https://vilua.atlassian.net/browse/SHAD-338

### Context
Emitting 2 events (Only when the user is authenticated):
- HOST_APP_OPEN
- RECCO_SDK_OPEN

### Media
https://github.com/SignificoHealth/recco-android-sdk/assets/11177510/b01638b8-2e65-4213-b27a-c2eeddecb57b

### CODE REVIEW

File: api/ui/src/main/kotlin/com/recco/api/ui/HostAppTrackEventsLifecycleObserver.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    private val metricRepository: MetricRepository
+) : DefaultLifecycleObserver {
+
+    init {
+        MainScope().launch {
+            ProcessLifecycleOwner.get()
+                .lifecycle
+                .addObserver(this@HostAppTrackEventsLifecycleObserver)
+        }
+    }
```
Comment (chemarubio) [inline api/ui/src/main/kotlin/com/recco/api/ui/HostAppTrackEventsLifecycleObserver.kt:26]:
```suggestion
) : DefaultLifecycleObserver, CoroutineScope by MainScope() {

    init {
        launch {
            ProcessLifecycleOwner.get()
                .lifecycle
                .addObserver(this@HostAppTrackEventsLifecycleObserver)
        }
    }
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 ## [1.0.0] - 2023-09-27 - YYYY-MM-DD
 
+### Added 
+- Send new metric events HOST_APP_OPEN & RECCO_SDK_OPEN https://vilua.atlassian.net/browse/SHAD-338
```
Comment (chemarubio) [inline CHANGELOG.md:6]:
```suggestion
- Send new metric events HOST_APP_OPEN & RECCO_SDK_OPEN [SHAD-338](https://vilua.atlassian.net/browse/SHAD-338)
```

File: internal/core/repository/src/main/kotlin/com/recco/internal/core/repository/MetricRepository.kt [hunk 3]
```diff
@@ -13,17 +14,16 @@ import javax.inject.Singleton
...
-                AppUserMetricEventDTO(
-                    category = AppUserMetricCategoryDTO.DASHBOARD_SCREEN,
-                    action = AppUserMetricActionDTO.VIEW
-                )
-            )
+    fun logEvent(event: AppUserMetricEvent) {
+        authCredentials.userId?.let {
+            appScope.launch {
+                runCatching { metricApi.logEvent(event.asDTO()) }
+                    .onFailure { logger.e(it) }
```
Comment (chemarubio) [inline internal/core/repository/src/main/kotlin/com/recco/internal/core/repository/MetricRepository.kt:25]:
```suggestion
                    .onFailure { logger::e }
```

File: internal/core/repository/src/main/kotlin/com/recco/internal/core/repository/mapper/MetricMapper.kt [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+internal fun AppUserMetricCategory.asDTO() = when (this) {
+    AppUserMetricCategory.USER_SESSION -> AppUserMetricCategoryDTO.USER_SESSION
+    AppUserMetricCategory.DASHBOARD_SCREEN -> AppUserMetricCategoryDTO.DASHBOARD_SCREEN
+}
+
+internal fun AppUserMetricEvent.asDTO() = AppUserMetricEventDTO(
+    category = this.category.asDTO(),
+    action = this.action.asDTO(),
+    value = this.valueEvent
+)
```
Comment (chemarubio) [inline internal/core/repository/src/main/kotlin/com/recco/internal/core/repository/mapper/MetricMapper.kt:25]:
```suggestion
internal fun AppUserMetricEvent.asDTO() = AppUserMetricEventDTO(
    category = category.asDTO(),
    action = action.asDTO(),
    value = valueEvent
)
```

**Review by VictorAlbertos — APPROVED** [review]
I think it is ready to merge, if you consider the changes suggested by @chemarubio please @robertofrontado make sure to test again 🙏

**Review by chemarubio — APPROVED** [review]
🚀

**Pull request merged**